circle.yml

```diff
@@ -379,64 +331,138 @@ jobs:
         steps:
             - checkout
             - halt-if-docker-image-exists:
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
+                  imageName: cypress/included:<< parameters.dockerTag >>
             - run:
-                  name: building Docker image << parameters.dockerName >>:<< parameters.dockerTag >>
+                  name: building Docker image cypress/included:<< parameters.dockerTag >>
                   command: |
-                      docker build -t << parameters.dockerName >>:<< parameters.dockerTag >> .
+                      docker build -t cypress/included:<< parameters.dockerTag >> --platform << parameters.platformArg >> .
                   working_directory: included/<< parameters.dockerTag >>
 
             - test-included-image-versions:
                   cypressVersion: << parameters.dockerTag >>
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
+                  imageName: cypress/included:<< parameters.dockerTag >>
 
             - test-included-image:
                   cypressVersion: << parameters.dockerTag >>
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
-
+                  imageName: cypress/included:<< parameters.dockerTag >>
             - halt-on-branch
-            - docker-push:
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
 
+    push-images:
+        ## the multi-arch images have been built and tested by previous jobs, so we're safe to blindly build+push
+        machine:
+            image: ubuntu-2004:202111-02
+        parameters:
+            dockerName:
+                type: string
+                description: Image name to build
+            dockerTag:
+                type: string
+                description: Image tag to build, should match Cypress version, like "3.8.1"
+            workingDirectory:
+                type: string
+                description: Directory with the Dockerfile
+            buildxArches:
+                type: string
+                description: List of architectures to pass to `docker buildx --platform`
+        steps:
+            - checkout
+            - halt-if-docker-image-exists:
+                imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
+            - halt-on-branch
+            - run:
+                ## see https://docs.docker.com/desktop/multi-arch/
+                name: build + push multi-arch image to docker hub
+                working_directory: << parameters.workingDirectory >>
+                command: |
+                    ## install QEMU, pulled from https://github.com/docker/setup-qemu-action/blob/4620c11b701e878e3f2b7aa142927d0646857fd9/src/main.ts
+                    ## avoids https://github.com/docker/buildx/issues/499
+                    BINFMT_IMAGE=tonistiigi/binfmt:latest@sha256:85683def11494bc0055d98dd1081e73cb5e70bbc1ae532be3d12cf054e2b11f1
```

Comment:
Are you able to add notes in the repo readme on how often and what testing is needed (if any) to maintain this image version? I am curious, why not use the existing script?

Reply:
The cited script is part of a GitHub Action, I didn't want to pull in all of that complexity to reverse-engineer running a GH Action from CircleCI, the `binfmt` Docker image just needs to run with `--privileged` to enable multi-arch builds, it doesn't need much more complexity than that.

The GH Action uses `latest`, it's pinned to a digest here because I don't like the idea of pulling in non-deterministic code in our CI.

Given this, is there specific documentation you'd like to see? I'd prefer to add it here in a comment if so, I think this stands a better chance of being discovered than a blurb in CONTRIBUTING.md.

Reply:
Sounds good. Docs on how often and any type of testing around this would be helpful, unless there is official docs around this vs pointing to a script we don't want to use.

Reply:
Added some in-line comments and general documentation here: [91bf5c1](https://github.com/cypress-io/cypress-docker-images/pull/700/commits/91bf5c1dc898f2d0a075f71c87a1976a86952b10)